vertx-service-discovery-backend-consul/src/test/java/io/vertx/servicediscovery/backend/consul/ConsulBackendTest.java

```diff
@@ -0,0 +1,54 @@
+package io.vertx.servicediscovery.backend.consul;
+
+import io.vertx.core.Vertx;
+import io.vertx.ext.consul.ConsulClient;
+import io.vertx.ext.consul.ConsulContext;
+import io.vertx.ext.consul.ConsulTestSuite;
+import io.vertx.ext.consul.impl.ConsulClientImpl;
+import io.vertx.servicediscovery.spi.ServiceDiscoveryBackend;
+import io.vertx.servicediscovery.spi.ServiceDiscoveryBackendTest;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import java.util.function.Function;
+
+/**
+ * @author <a href="mailto:cafeinoman@openaliasbox.org>Francois Delalleau</a>
+ */
+public class ConsulBackendTest extends ServiceDiscoveryBackendTest {
+
+  private static Function<Vertx, ConsulContext> ctxFactory;
+  private ConsulContext ctx;
+
+  @BeforeClass
+  static public void startConsul() throws Exception {
+    ctxFactory = vertx -> new ConsulContext(
```

Comment:
`ConsulContext` is internal class and it not intended to be public due the future changes, removes etc. Please use `embedded-consul` API directly